persistence/leveldb/src/test/java/org/infinispan/persistence/leveldb/config/ConfigurationTest.java

```diff
@@ -65,18 +69,22 @@ public void testConfigBuilder() {
    }
 
    public void testXmlConfig() throws IOException {
-      try {
-         EmbeddedCacheManager cacheManager = new DefaultCacheManager("config/leveldb-config-" +
-                 LevelDBStoreConfiguration.ImplementationType.AUTO.toString().toLowerCase() + ".xml");
-
-         Cache<String, String> cache = cacheManager.getCache("testCache");
-
-         cache.put("hello", "there 52 xml");
-         cache.stop();
-         cacheManager.stop();
-      } finally {
-         Util.recursiveFileRemove("/tmp/leveldb/52");
-      }
+      InputStream configSTream = ConfigurationTest.class.getResourceAsStream("/config/leveldb-config-auto.xml");
+      ConfigurationBuilderHolder configHolder = new ParserRegistry().parse(configSTream);
+
+      Configuration cacheConfig = configHolder.getNamedConfigurationBuilders().get("testCache").build();
+      assertFalse(cacheConfig.persistence().passivation());
+      assertEquals(cacheConfig.persistence().stores().size(), 1);
+
+      StoreConfiguration cacheLoaderConfig = cacheConfig.persistence().stores().get(0);
```

Comment:
Even though it may look awkward - grouping test preparation (`//given`), action (`//when`) and result validation (`//then`) makes a lot of sense to me. If you follow this pattern, the test tells you some story instead of going through some scenario randomly checking some parts...